### PR TITLE
Refo 26

### PR DESCRIPTION
**Pull request by 10RE**

### CODE REVIEW

File: src/components/Navbar.jsx [hunk 1]
```diff
@@ -1,58 +1,143 @@
...
+    bmMenu: {
+      background: "white",
+      padding: "2.5em 1.5em 0",
+      fontSize: "1.15em",
+    },
+    bmOverlay: {
+      background: "rgba(0, 0, 0, 0.3)",
+    },
+  };
+
```
Comment (makersmelx) [inline src/components/Navbar.jsx:125]: i would prefer to set style by adding tailwind `className` by here https://github.com/negomi/react-burger-menu#custom-id-andor-classnames

for color, if there is no place to set className,  please refer to https://tailwindcss.com/docs/customizing-colors

**Review by makersmelx — CHANGES_REQUESTED** [review]
please help to resolve the conflict first. After that, we may need to let Neha @nehakumari2307  take a review at the appearance.

File: src/components/Navbar.jsx [hunk 2]
```diff
@@ -1,58 +1,143 @@
...
+}
+
+export default function Navbar() {
+  const [width, setWidth] = useState(window.innerWidth);
+  const [open, setOpen] = useState(false);
+
+  useEffect(() => {
+    const handleResize = () => setWidth(window.innerWidth);
+    window.addEventListener("resize", handleResize);
+  });
```
[the remark this answers is not in the document]
Comment (10RE) [reply src/components/Navbar.jsx:93]: i don't get it, what do you mean by flex class @makersmelx

**Review by makersmelx — CHANGES_REQUESTED** [review]
In the mobile view, seems the menu occupies some of the spaces, which is not expected. the main content should occupy the full page.
image


In the PC view, when the user is not logged in, we have a blank on the left side, which is also not expected
image

File: src/components/Navbar.jsx [hunk 3]
```diff
@@ -1,58 +1,162 @@
...
-export default function Navbar() {
+import { ROLES } from "@/constants";
+import { useAuth } from "@/models";
+
+const buttonMap = {
+  [ROLES.ADMIN]: {
+    buttons: {
+      "Create Account": {
+        icon: <AddIcon />,
+        path: "",
```
Comment (makersmelx) [inline src/components/Navbar.jsx:19]: add a `todo` for pages that are not merged/created yet

**Review by makersmelx — CHANGES_REQUESTED** [review]

**Review by makersmelx — CHANGES_REQUESTED** [review]
please add a todo in the class related props of the hamburger menu
`todo: fix dark mode compatibility`

the gap is not expected
image

still not in the center
image

**Review by makersmelx — APPROVED** [review]

**Pull request merged**